plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java

```diff
@@ -431,13 +458,16 @@ private DeltaLakeTransactionLogEntry buildAddEntry(ConnectorSession session, Blo
             statsFieldIndex = 5;
         }
 
-        Optional<DeltaLakeParquetFileStatistics> parsedStats = Optional.ofNullable(getRowField(addEntryRow, statsFieldIndex + 1)).map(this::parseStatisticsFromParquet);
+        boolean partitionValuesParsedExists = addEntryRow.getUnderlyingFieldBlock(statsFieldIndex + 1) instanceof RowBlock && // partitionValues_parsed
```

Comment:
Do we need to check this for every position ? Seems like we should know this per file based on parquet file metadata (maybe it's possible to use io.trino.plugin.hive.ReaderPageSource#getReaderColumns).

Reply:
Agree with using Parquet metadata though getReaderColumns returns an empty list in this case. Preparing another PR.